docs/source/recipes/deploying-elyra-in-a-jupyterhub-endvironment.md

```diff
@@ -0,0 +1,121 @@
+<!--
+{% comment %}
+Copyright 2018-2020 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+# Deploying Elyra & JupyterHub in a Kubernetes environment
+
+This document goes over how to integrate Elyra into a self-service Jupyter Notebook platform
+managed by JupyterHub.
+
+If you are looking for details on how to install/configure JupyterHub, look at
+[Zero to JupyterHub with Kubernetes](https://zero-to-jupyterhub.readthedocs.io/en/latest/index.html)
+or this [Ansible script to deploy Elyra, JupyterHub, Jupyter Enterprise Gateway](https://github.com/lresende/ansible-kubernetes-cluster)
+
+![JupyterHub Deployment](https://raw.githubusercontent.com/lresende/ansible-kubernetes-cluster/master/docs/images/elyra-deployment-diagram.png)
+
+
+## Configuring JupyterHub Deployment
+
+In this case, JupyterHub is the recommended entry point for Elyra, as it will manage the user
+authorization and provisioning of individual Notebook servers for each user.
+
+JupyterHub is configured via a config.yaml file, and the following settings are required:
+
+ * Configure JupyterHub to use Elyra docker image when instantiating the notebook environment for each user
+
+```yaml
+singleuser:
+  defaultUrl: "/lab"
+  image:
+    name: elyra/elyra
+    # change to a specific release version as appropriated
+    tag: dev
+    # disable this in a production environment
+    pullPolicy: "Always"
+  storage:
+    dynamic:
+      storageClass: nfs-dynamic
+```
+
+## Deploying Jupyter Enterprise Gateway
+
+Jupyter Enterprise Gateway enables support for remote kernels in a Jupyter Deployment
+
+ * Deploying Jupyter Enterprise gateway
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/jupyter/enterprise_gateway/v2.1.1/etc/kubernetes/enterprise-gateway.yaml
+```
+
+## Connecting JupyterHub and Jupyter Enterprise Gateway
+
+Now that all components are deployed, we need to make them aware of each other via some
+additional environment variables.
+
+ * Flowing current user from JupyterHub to Jupyter Enterprise Gateway
+
+```yaml
```

Comment:
Should make aware that these settings need to go into that same JupyterHub `config.yaml` file. 
At this point the user should have already deployed `z2jh` so they will need to modify the config map. Should we provide instructions on doing so? Or better, move these instructions further up so the changes can be made prior to deployment?